ganga/GangaGUI/gui/routes.py

```diff
@@ -1,71 +1,1018 @@
+import pty
+import sys
+import datetime
 from functools import wraps
-from itertools import chain
-from flask import request, jsonify, render_template, flash
-from GangaGUI.gui import app
-from GangaGUI.gui.models import User
+from werkzeug.utils import secure_filename
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask import Flask, request, jsonify, render_template, flash, redirect, url_for, session, send_file
+from flask_login import login_user, login_required, logout_user, current_user, UserMixin
+from flask_sqlalchemy import SQLAlchemy
+from flask_login import LoginManager
+from flask_socketio import SocketIO
+from GangaGUI.gui.config import Config
+
+# ******************** Initialisation of Flask App for GUI ******************** #
+
+
+# GUI Flask App and set configuration from ./config.py file
+gui = Flask(__name__)
+gui.config.from_object(Config)
+
+# Database object which is used to interact with the "gui.sqlite" in gangadir/gui folder
+# NOTE: IT HAS NO RELATION WITH THE GANGA PERSISTENT DATABASE
+db = SQLAlchemy(gui)
+
+# Login manage for the view routes
+login = LoginManager(gui)
+login.login_view = "login"
+login.login_message = "Please Login to Access this Page."
+login.login_message_category = "warning"
+
+# For websocket, for communication between frontend and backend
+socketio = SocketIO(gui)
+
+
+# ******************** The user class for database and authentication ******************** #
+
+# ORM Class to represent Users - used to access the GUI & API resources
+class User(UserMixin, db.Model):
+    __tablename__ = "users"
+    id = db.Column(db.Integer, primary_key=True)
+    public_id = db.Column(db.String(64), unique=True)
+    user = db.Column(db.String(32), unique=True)
+    password_hash = db.Column(db.String(64))
+    role = db.Column(db.String(32))
+    pinned_jobs = db.Column(db.Text)
+
+    def store_password_hash(self, password: str):
+        self.password_hash = generate_password_hash(password)
+
+    def verify_password(self, password: str) -> bool:
+        return check_password_hash(self.password_hash, password)
+
+    def generate_auth_token(self, expires_in_days: int = 5) -> str:
+        return jwt.encode(
+            {"public_id": self.public_id, "exp": datetime.datetime.utcnow() + datetime.timedelta(days=expires_in_days)},
+            gui.config["SECRET_KEY"], algorithm="HS256")
+
+    def __repr__(self):
+        return "User {}: {} (Public ID: {}, Role: {})".format(self.id, self.user, self.public_id, self.role)
+
+
+# User Loader Function for Flask Login
+@login.user_loader
+def load_user(user_id):
+    return User.query.get(int(user_id))
+
+
+# ******************** Global Variables ******************** #
+
+
+# Colors showed for different job statuses in the GUI based on Bootstrap CSS
+status_color = {
+    "new": "info",
+    "completed": "success",
+    "failed": "danger",
+    "running": "primary",
+    "submitted": "secondary",
+    "killed": "warning"
+}
```

Comment:
The status will be shown, it just won't have a color. It will be shown a plain text instead of green background (which you see in completed jobs) or such.